### PR TITLE
Add checks for better map assertions

### LINKED ISSUE #3: Detect uses of Map.keys property extraction for checking if a key does not exist.
willowtreeapps/assertk#502 added `doesNotContainKey` assertions for maps. This should be encouraged over `keys` property extraction for consistent error messages.

This should be reported as a warning and should include a quick fix.

### PR DESCRIPTION
**Pull request by jzbrooks**
Fixes #3

### CODE REVIEW

**Comment by jzbrooks** [comment]
@evant, if you have some time to sanity check this conceptually, I'd appreciate it!

File: checks/src/test/kotlin/com/jzbrooks/assertk/lint/checks/MapAssertionDetectorTest.kt [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+            """.trimIndent()
+
+        lint().files(kotlin(code), kotlin(assertkStub), kotlin(assertkCollectionStub)).run().expect(
+            """src/clean/TestingTesting.kt:11: Warning: Use Assert.doesNotContainKey to assert absence [KeySetAbsentAssertion]
+        assertThat(map.keys).doesNotContain("")
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+0 errors, 1 warnings""",
+        )
+    }
+
```
Comment (erawhctim) [inline checks/src/test/kotlin/com/jzbrooks/assertk/lint/checks/MapAssertionDetectorTest.kt:270]: Do you care about usages like `map.entries.map { it.key }`?

Comment (jzbrooks) [reply checks/src/test/kotlin/com/jzbrooks/assertk/lint/checks/MapAssertionDetectorTest.kt:270]: Maybe. Have you seen that call before in the wild?

It strikes me as odd to iterate the entries to read a set that the map is probably already maintaining internally, but if people do that for whatever reason…

Comment (erawhctim) [reply checks/src/test/kotlin/com/jzbrooks/assertk/lint/checks/MapAssertionDetectorTest.kt:270]: Ha, honestly I'm not sure; I was mostly trying to find edge cases related to keys. 

I think I've written it before, but probably incorrectly 🙂

Comment (jzbrooks) [reply checks/src/test/kotlin/com/jzbrooks/assertk/lint/checks/MapAssertionDetectorTest.kt:270]: I'm going to merge this, but I created #11 to track interest in something like that over time. I explained a little of my rationale there.
